internal/security/sensitive_string.go

```diff
@@ -0,0 +1,54 @@
+package security
+
+import (
+	"encoding/json"
+)
+
+const redactionPlaceholder = "***"
+
+// SensitiveString is a type for handling sensitive information securely.
+// This helps to achieve the following goals:
+//  1. Prevent accidental logging of sensitive information.
+//  2. Provide controlled visibility (e.g., redacted output for String() or MarshalJSON()).
+//  3. Enable controlled access to the sensitive value when needed.
+type SensitiveString struct {
+	value string
+}
+
+// NewSensitiveString creates a new SensitiveString
+func NewSensitiveString(value string) SensitiveString {
+	return SensitiveString{value: value}
+}
+
+// String provides a redacted version of the sensitive string
+func (s *SensitiveString) String() string {
+	if s.value == "" {
+		return ""
+	}
+	return redactionPlaceholder
+}
+
+// MarshalJSON ensures that sensitive strings are redacted when marshaled to JSON
+func (s *SensitiveString) MarshalJSON() ([]byte, error) {
+	return json.Marshal(s.String())
+}
```

Comment:
> On the other side, it will automatically prevent us from showing tokens to the desktop/mobile

Wdym by tokens? :thinking: 

> but might there be any cases where we actually NEED to pass some saved secrets from status-go to the client?

Even if there is a case like this, we could simply not use `SensitiveString` for such fields or introduce custom `MarshalJSON` where it is transformed to normal string with `Reveal` method.

Reply:
> Wdym by tokens? 🤔

Like Infura tokens e.g. But I mean any secrets.

> Even if there is a case like this, we could simply not use SensitiveString for such fields or introduce custom MarshalJSON where it is transformed to normal string with Reveal method.

Sure, the questions is if we know of such cases now, to prevent issues 🙂